src/dataflow/src/render/mod.rs

```diff
@@ -674,26 +674,67 @@ pub(crate) fn build_dataflow<A: Allocate>(
                     ))
                     .expect("Sink source collection not loaded");
 
-                // TODO(frank): consolidation is only required for a collection,
-                // not for arrangements. We can perform a more complicated match
-                // here to determine which case we are in to avoid this call.
-                use differential_dataflow::operators::consolidate::Consolidate;
-                let collection = collection.consolidate();
+                // TODO(frank): Consolidation can be avoided for arranged inputs.
+
+                // These steps perform consolidation, but do not devolve the stream
+                // from batches to individual updates. That can happen for sinks that
+                // require e.g. updates shuffled, but for sinks that do not require
```

Comment:
Is there a typo in this comment? What does it mean for a sink to require that an update is shuffled? Should "shuffled" be "sorted"?

Reply:
I think what I meant to say (tbh, not certain) is that some sink operators may require the records to be shuffled among the workers, perhaps for example to place all of the updates at one location. The "That can happen" was meant to be "it is still possible to perform this". Does that make it clearer?